File: ui/hub-mk3-ui/src/store/orm/program-sequence-binding-meme.js
// Copyright (c) XJ Music Inc. (https://xj.io) All Rights Reserved.
import {attr, fk, Model} from "redux-orm";

class ProgramSequenceBindingMeme extends Model {
  toString() {
    return `ProgramSequenceBindingMeme: ${this.name}`;
  }

  // Declare any static or instance methods you need.
}

ProgramSequenceBindingMeme.modelName = 'ProgramSequenceBindingMeme';

// Declare your related fields.
ProgramSequenceBindingMeme.fields = {
  id: attr(),
  name: attr(),
  programId: fk({
    to: 'Program',
    as: 'program',
    relatedName: 'sequenceBindingMemes',
  }),
  programSequenceBindingId: fk({
    to: 'ProgramSequenceBinding',
    as: 'programSequenceBinding',
    relatedName: 'memes',
  }),
};

export default ProgramSequenceBindingMeme;

File: engine/src/content/ProgramSequenceBindingMeme.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include "xjmusic/content/ProgramSequenceBindingMeme.h"

using namespace XJ;

std::set<std::string> ProgramSequenceBindingMeme::getNames(const std::set<ProgramSequenceBindingMeme> &programSequenceBindingMemes) {
  std::set<std::string> names;
  for (const auto &programSequenceBindingMeme: programSequenceBindingMemes) {
    names.insert(programSequenceBindingMeme.name);
  }
  return names;
}

